plasmapy/formulary/drifts.py

```diff
@@ -14,6 +16,67 @@
 from plasmapy.utils.decorators import validate_quantities
 
 
+@validate_quantities
+def diamagnetic_drift(dp: u.Pa / u.m, B: u.T, n: u.m ** (-3), q: u.C) -> u.m / u.s:
+    r"""
+    Calculate the diamagnetic fluid perpendicular drift.
+
+    **Aliases:** `vd_`
+
+    Parameters
+    ----------
+    dp : ~astropy.units.Quantity
+        Pressure gradient vector
+    B  : ~astropy.units.Quantity
+        Magnetic field vector
+    n  : ~astropy.units.Quantity
+        Number density
+    q  : ~astropy.units.Quantity
+        Particle charge
+
+    Returns
+    -------
+    v: ~astropy.units.Quantity
+        Drift velocity, in m/s
+
+    Examples
+    --------
+    >>> import astropy.units as u
+    >>> import numpy as np
+    >>> dp = np.array([0, -1, 0]) * u.Pa / u.m
+    >>> b = np.array([0, 0, 1]) * u.T
+    >>> n = 1 / u.m ** 3
+    >>> q = 1 * u.C
+    >>> diamagnetic_drift(dp, b, n, q)
+    <Quantity [ 1., -0., -0.] m / s>
+
+    Notes
+    -----
+    The diamagnetic drift is given by
+
+    .. math::
+
+        \vec{v} = -\frac{ \nabla p \times \vec{B} }{ q n |B|^2 }
+
+    This is the velocity component of a fluid element perpendicular to the
+    magnetic field.
+
+    References
+    ----------
+    - Chen, Introduction to Plasma Physics and Controlled Fusion, 3.65
+
+    """
+
+    # np.cross drops units right now, thus this hack: see
+    # https://github.com/PlasmaPy/PlasmaPy/issues/59
+    cross = np.cross(dp.si.value, B.si.value) * dp.unit * B.unit
+    return -cross / q / n / (B * B).sum(-1)
```

Comment:
Interestingly, I just checked:

```ipython
[ins] In [2]: a = u.Quantity([1, 0, 0], u.m)

[nav] In [3]: b = u.Quantity([0, 1, 0], u.s)

[ins] In [4]: np.cross(a, b)
Out[4]: <Quantity [0., 0., 1.] m s>
```

seems to be working with numpy 1.19.4 and astropy 4.2 (which, come to think of it, happens to be 10% of the answer to the universe, life and anything...). So I wonder - will test break if we remove the hack?

```suggestion
    cross = np.cross(dp, B)
    return -cross / q / n / (B * B).sum(-1)
```